File: boyorgirl.cpp
#include <bits/stdc++.h>
using namespace std;

bool already(vector<string> v, string str) {
	int count = 0;
	for (int i = 0; i < v.size(); i++) {
		if (v[i] == str) count++;
	}
	if (count > 0) {
		return true;
	} else {
		return false;
	}
}

int main() {
	string user; vector<string> vec;
	cin >> user;
	for (int i = 0; i < user.length(); i++) {
		if (already(vec, user[i]) == false) {
			vec.push_back(user[i]);
		}
	}
	if (vec.size() % 2 == 0) {
		cout << "CHAT WITH HER!\n";
	} else {
		cout << "IGNORE HIM!\n";
	}
}

File: boyorgirl.java
import java.util.*;

public final class boyorgirl {
	public static void main(String[] args) {
		TreeSet<Character> letters = new TreeSet<Character>();
		Scanner sc = new Scanner(System.in);
		char[] thing = (sc.nextLine()).toCharArray();
		for (char i : thing) {
			letters.add(i);
		}
		if (letters.size() % 2 == 0) {
			System.out.println("CHAT WITH HER!");
		} else {
			System.out.println("IGNORE HIM!");
		}
	}
}